torch/_refs/__init__.py

```diff
@@ -1552,6 +1579,64 @@ def reshape(a: TensorLikeType, shape: ShapeType) -> TensorLikeType:
     return _reshape_view_helper(a, shape, allow_copy=True)
 
 
+def rot90(a: TensorLikeType, k: int = 1, dims: DimsSequenceType = (0, 1)) -> TensorLikeType:
+    """Reference implementation of :func:`torch.rot90`."""
+    if len(dims) != 2:
```

Comment:
I think this needs to canonicalize the dims upfront with utils.canonicalize_dims (which wraps dims specified as negative numbers)

Reply:
Initially I thought that `canonicalize_dims` is not needed here because `dims` is passed to `flip` and `transpose` which themselves call `canonicalize_dims` taking care of negative numbers. But then I see now that the "out of range" checks are unnecessarily duplicated in this function.

Reply:
Eventually we should investigate an error context so even if an error is thrown from another operation it cites the user-facing entrypoint as expected.

However, we don't have an error context today, so I would suggest duplicating error checks for the moment both for message quality and for code simplicity -- because it can sometimes be tricky to reason about what other operations will or will not validate